kong/plugins/cors/handler.lua

```diff
@@ -56,8 +58,25 @@ end
 
 local function configure_credentials(ngx, conf)
   if ngx.ctx.cors_allow_all then
-    ngx.header["Access-Control-Allow-Credentials"] = "false"
+    local allow_credentials = false
+
+    -- If it's catch all, then match the Origin requested by the client
+    -- but only if the Origin URL is valid, and only if credentials=true
```

Comment:
i.e., this also "strips" important parts of the URL that would be needed for the client to verify the result of the ACAO header, but now, the client will only receive scheme + host (no authority, no port).